pkgs/by-name/db/dbvisualizer/package.nix

```
@@ -0,0 +1,61 @@
+{
+  lib,
+  makeDesktopItem,
+  makeWrapper,
+  stdenv,
+  temurin-jre-bin-17,
+}:
+
+let
+  pname = "dbvisualizer";
+  version = "24.2.2";
+  underscoreVersion = lib.concatStringsSep "_" (lib.splitString "." version);
+
+  desktopEntry = makeDesktopItem {
+    name = pname;
+    desktopName = "DB Visualizer";
+    tryExec = "dbvis";
+    exec = "dbvis";
+    icon = "dbvisualizer";
+    categories = [ "Development" ];
+    terminal = false;
+    keywords = [
+      "Database"
+      "SQL"
+    ];
+  };
+in
+stdenv.mkDerivation {
+  inherit pname version;
+
+  src = fetchTarball {
+    url = "https://www.dbvis.com/product_download/dbvis-${version}/media/dbvis_linux_${underscoreVersion}.tar.gz";
+    sha256 = "sha256:17kbf444n21jawr55w4lr0l216rdjf4w3cykj04i1h2hqbjswvp8";
+  };
+
+  buildInputs = [ makeWrapper ];
+
+  strictDeps = true;
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp -a . $out
+
+    mkdir -p $out/share/
+    cp -a ${desktopEntry}/share/applications $out/share
+
+    mkdir -p $out/share/icons/hicolor/128x128/apps
+    ln -s $out/resources/images/tool/dbvis-icon128x128.png \
+      $out/share/icons/hicolor/128x128/apps/dbvisualizer.png
+
+    ln -s $out/dbvis $out/bin
+    wrapProgram $out/bin/dbvis --set INSTALL4J_JAVA_HOME ${temurin-jre-bin-17}
+  '';
+
+  meta = {
+    description = "The universal database tool";
+    homepage = "https://www.dbvis.com/";
+    maintainers = with lib.maintainers; [ boldikoller ];
+    license = lib.licenses.unfree;
+  };
```

```suggestion
    mainProgram = "dbvis";
  };
```

Would you add `mainProgram` ?

---

Yes, thank you for pointing out.